client/components/swipeable/index.js

```diff
@@ -0,0 +1,212 @@
+		}, [ updateLayout ] );
+
+		// fire when the `currentPageIndex` parameter changes
+		useEffect( () => {
+			if ( ! enabled ) {
+				return;
+			}
+
+			savedUpdateLayout.current();
+		}, [ enabled, currentPageIndex ] );
+
+		// fire when the window resizes
+		useEffect( () => {
+			if ( ! enabled ) {
+				return;
+			}
+
+			const onResize = () => savedUpdateLayout.current( false );
+			window.addEventListener( 'resize', onResize );
+			window.addEventListener( 'orientationchange', onResize );
+			return () => {
+				window.removeEventListener( 'resize', onResize );
+				window.removeEventListener( 'orientationchange', onResize );
+			};
+		}, [ enabled ] );
+	}
+
+	const updateEnabled = hasDynamicHeight && numPages > 1;
+
+	useUpdateLayout( updateEnabled, currentPage, ( hasTransitionDuration = true ) => {
+		const offset = getOffset( currentPage );
+		const transform = {
+			transform: `translate3d(${ offset }px, 0px, 0px)`,
+			transitionDuration: hasTransitionDuration ? `300ms` : `0ms`,
+		};
+
+		let height = { height: null };
+		const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
+
+		if ( targetHeight && pagesStyle?.height !== targetHeight ) {
+			height = { height: null };
+		}
+
+		setPagesStyle( { ...pagesStyle, ...height, ...transform } );
+	} );
+
+	useEffect( () => {
+		if ( currentPage >= numPages ) {
+			onPageSelect( numPages - 1 );
+		}
+	}, [ numPages, currentPage, onPageSelect ] );
+
+	const getDragPosition = ( event ) => {
+		if ( event.hasOwnProperty( 'clientX' ) ) {
+			return { x: event.clientX, y: event.clientY };
+		}
+
+		if ( event.targetTouches[ 0 ] ) {
+			return { x: event.targetTouches[ 0 ].clientX, y: event.targetTouches[ 0 ].clientY };
+		}
+
+		const touch = event.changedTouches[ 0 ];
+		return { x: touch.clientX, y: touch.clientY };
+	};
+
+	const handleDragStart = ( event ) => {
+		const position = getDragPosition( event );
+		setDragStartData( position );
+		setPagesStyle( { ...pagesStyle, transitionDuration: `0ms` } ); // Set transition Duration to 0 for smooth dragging.
+		setIsDragging( true );
+	};
+
+	const handleDragEnd = ( event ) => {
+		if ( ! isDragging ) {
+			return; // End early if we are not dragging any more.
+		}
+
+		const THRESHOLD_OFFSET = 100; // Number of pixels to travel before we trigger the slider to move to the desired slide.
+		const dragPosition = getDragPosition( event );
+		const delta = dragPosition.x - dragStartData.x;
+
+		const hasMetThreshold = Math.abs( delta ) > THRESHOLD_OFFSET;
+
+		let newIndex = currentPage;
+		if ( delta < 0 && hasMetThreshold && numPages !== currentPage + 1 ) {
+			newIndex = currentPage + 1;
+		}
+
+		if ( delta > 0 && hasMetThreshold && currentPage !== 0 ) {
+			newIndex = currentPage - 1;
+		}
+		const offset = getOffset( newIndex );
+		setPagesStyle( {
+			transform: `translate3d(${ offset }px, 0px, 0px)`,
+			transitionDuration: `300ms`,
+		} );
+		onPageSelect( newIndex );
+		setDragStartData( {} );
```

Comment:
Why using an empty object instead of `null` or `undefined` ? (Same question for this state's initial value)

Reply:
Good Idea. I was able to remove the isDragging state as well.